### PR TITLE
Add detailed git workflow

### PR DESCRIPTION
**Pull request by JerryFireman**
Part of #747

---

# Description

@terenc3t requested this pull request.

Updated CONTRIBUTING.md to add more detailed GIt and Github workflow.

Also see: https://github.com/prysmaticlabs/prysm/pull/747

### CODE REVIEW

File: k8s/beacon-chain/beacon-chain.deploy.yaml [hunk 1]
```diff
@@ -14,24 +13,17 @@ spec:
...
-        prometheus.io/scrape: 'true'
-        prometheus.io/port: '9090'
     spec:
       priorityClassName: production-priority
       containers:
         - name: beacon-chain
           image: gcr.io/prysmaticlabs/prysm/beacon-chain:latest
           args: 
-            - "--web3provider=ws://geth-nodes.pow.svc.cluster.local:8546"
+            - "--web3provider=ws://$(GETH_NODES_SERVICE_HOST):$(GETH_NODES_SERVICE_PORT_WS)"
```
Comment (prestonvanloon) [inline k8s/beacon-chain/beacon-chain.deploy.yaml]: Please revert these changes

File: .idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (prestonvanloon) [inline .idea/misc.xml]: Please do not include intellij directory.

**Comment by prestonvanloon** [comment]
I updated your branch and the description. I hope thats OK!

**Comment by JerryFireman** [comment]
@terenc3t Thanks very much Terence! Great working with you guys!
